### PR TITLE
Make Vault non-recoverable

### PR DESCRIPTION
**Pull request by sohkai**

### CODE REVIEW

File: apps/vault/test/vault.js [hunk 1]
```diff
@@ -128,9 +134,90 @@ contract('Vault app', (accounts) => {
...
+
+      vaultId = hash('vault.aragonpm.test')
+      const APP_BASE_NAMESPACE = await kernelBase.APP_BASES_NAMESPACE()
+
+      const vaultReceipt = await kernel.newAppInstance(vaultId, vaultBase.address)
+      const vaultProxyAddress = getEvent(vaultReceipt, 'NewAppProxy', 'proxy')
+      vault = Vault.at(vaultProxyAddress)
+    })
+
+    it('forwards funds', async () => {
```
Comment (sohkai) [inline apps/vault/test/vault.js]: @izqui I wasn't quite sure what this test that you had scaffolded from before was supposed to do.

Comment (izqui) [reply apps/vault/test/vault.js]: Trying to come up with an excuse but I don't remember at all tbf :)

File: apps/vault/test/vault.js [hunk 2]
```diff
@@ -128,9 +134,90 @@ contract('Vault app', (accounts) => {
...
+      it('set up the default vault correctly to recover ETH from the kernel', async () => {
+        await kernel.sendTransaction({ value: 1, gas: 31000 })
+        assert.equal((await getBalance(kernel.address)).valueOf(), 1, 'kernel should have 1 balance')
+
+        await kernel.transferToVault(ETH)
+        assert.equal((await getBalance(kernel.address)).valueOf(), 0, 'kernel should have 0 balance')
+        assert.equal((await getBalance(defaultVault.address)).valueOf(), 1, 'default value should have 1 balance')
+
+        // Now make sure that if we try this on the vault, it fails
+        await vault.sendTransaction({ value: 1, gas: 31000 })
```
Comment (izqui) [inline apps/vault/test/vault.js]: What is this testing?

Comment (sohkai) [reply apps/vault/test/vault.js]: Oops, forgot to delete it 👍

File: apps/vault/contracts/Vault.sol [hunk 3]
```diff
@@ -94,4 +94,12 @@ contract Vault is AragonApp {
...
         }
     }
+
+    /**
+    * @dev Disable recovery escape hatch, as it could be used
+    *      maliciously to transfer funds away from the vault
+    */
+    function allowRecoverability(address token) public view returns (bool) {
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (bingen) [reply apps/vault/contracts/Vault.sol]: Oh, thanks, I forgot about it. Although it's kind of obvious, wouldn't it be good to have a unit test for it? At least it's an easy one to increase coverage ;-)

[the remark this answers is not in the document]
Comment (bingen) [reply apps/vault/contracts/Vault.sol]: Oh, thanks, I forgot about it.

**Pull request merged**